sdk/core/Azure.Core/src/Shared/MutableJsonDictionary.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+
+#nullable enable
+
+namespace Azure.Core.Json
+{
+    // TODO: this should use Value when availble, to avoid boxing value types.
+    internal readonly struct MutableJsonDictionary<T> : IDictionary<string, T>
```

Comment:
what is this type for?

Reply:
It is one approach to implementing the `IDictionary` interface in model types via MutableJsonDocument.

Reply:
I think we have to keep the helper type in order to make it templated.  It needs to be templated to satisfy the contract our generated models have, and I don't think we want a templated version of MutableJsonElement just to implement the collection interfaces.  That feels strange - it would be saying - all array elements or dictionary items for the whole JsonElement are of type T.